net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+        }
+    }
+
+    proc install_initial_configuration {f_or_d} {
+        if { [variant_isset "initialize"]
+             && [file exists ${f_or_d}]
+            } {
+            delete ${f_or_d}.previous
+            move \
+                ${f_or_d} \
+                ${f_or_d}.previous
+        }
+        if { [variant_isset "initialize"]
+             || ![file exists ${f_or_d}]
+            } {
+            if { [file isfile ${f_or_d}.macports] } {
+                xinstall -m 0644 \
+                    ${f_or_d}.macports \
+                    ${f_or_d}
+            } elseif { [file isdirectory ${f_or_d}.macports] } {
+                xinstall -m 0755 -d ${f_or_d}
+                foreach f [glob -nocomplain ${f_or_d}.macports/*] {
+                    xinstall -m 0644 ${f} \
+                        ${f_or_d}/[file tail ${f}]
+                }
+            }
+        }
+    }
+
+    # calenderserver configuration and nginx reverse proxy
+    foreach f_or_d {
+        Config/calendarserver.plist
+        etc/nginx.conf
+    } {
+        install_initial_configuration \
+            ${calendarserverdir}/Library/CalendarServer/${f_or_d}
+    }
+
+    # configure all template files with local settings
+    foreach d_or_f {
+        Config
+        etc
+        } {
+        fs-traverse f ${calendarserverdir}/Library/CalendarServer/${d_or_f} {
+            if { [file isfile ${f}]
+                 && ![string match "*.macports" ${f}]
+                 && [string match "text/*" \
+                         [lindex [exec /usr/bin/file --mime-type ${f}] end]]
+             } then {
+                foreach cmd [list \
+                     "s|@PREFIX@|${prefix}|g" \
+                     "s|@host@.@domain@.@tld@|${fullhost}|g" \
+                     "s|@domain@.@tld@|${domaintld}|g" \
+                     "s|@host@|${host}|g" \
+                     "s|@domain@|${domain}|g" \
+                     "s|@tld@|${tld}|g" \
+                     "s|@TLS_CERTIFICATE_NAME@|${tls_certificate_name}|g" \
+                     "s|@POSTGRESQL9_VERSION_NICKNAME@|${postgresql9_version_nickname}|g" \
+                    ] {
+                    reinplace -q ${cmd} ${f}
+                }
+            }
+        }
+    }
+
+    # Add launchd.plist keys to MacPorts launchdaemons
+
+    # org.macports.calendarserver
+    system -W \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver \
+        "/usr/bin/plutil -insert EnvironmentVariables -xml \
+            \"<dict> \
+                <key>PYTHON</key> \
+                <string>${calendarserverdir}/Library/CalendarServer/${ccsname}/bin/python</string> \
+                <key>PATH</key> \
+                <string>${calendarserverdir}/Library/CalendarServer/${ccsname}/bin:${calendarserverdir}/Library/CalendarServer/${ccsname}/virtualenv/bin:${prefix}/bin:${prefix}/sbin:/usr/bin:/usr/sbin:/bin:/sbin</string> \
```

Comment:
Why can't you use python shipped with MacPorts?

Reply:
Couldn’t get it to build, couldn’t get it to work. Native Python works, so I’m using it.

By the way, this is a `virtualenv` Python, so it could just as well be MacPorts Python, if ccs-calendarserver worked with MacPorts Python, which I haven’t been able to get working.

Reply:
This is Python-agnostic, and works with MacPorts Python.